tc/aten/aten_compiler_new_api-inl.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "tc/aten/aten_compiler.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "tc/aten/aten.h"
+#include "tc/core/compiler.h"
+#include "tc/core/tc_executor.h"
+#include "tc/core/tensor.h"
+
+using tc::DLConstTensorUPtr;
+using tc::DLTensorUPtr;
+
+namespace tc {
+namespace aten {
+inline std::vector<DLTensorUPtr> inferOutputTensorInfo(
+    const std::string& tc,
+    const std::string& entryPoint,
+    const std::vector<at::Tensor>& inputs) {
+  auto parsedTcs = tc::detail::parse(tc);
+  if (parsedTcs.count(entryPoint) != 1u) {
+    CHECK_GE(parsedTcs.size(), 1u)
+        << "No TC was parsed, should have thrown earlier";
+    throw lang::ErrorReport(parsedTcs.begin()->second)
+        << "\nattempting to access undefined entryPoint: " << entryPoint;
+  }
+  auto inputDLTensors = toDLConstTensors(inputs);
+  return makeDLTensorVector(detail::inferOutputTensorInfo(
+      parsedTcs.at(entryPoint), extractRawPtrs(inputDLTensors)));
+}
+
+inline std::vector<at::Tensor> prepareOutputs(
```

Comment:
why do we want to inline these relatively long functions?

Reply:
was trying to avoid creating yet another library (tc_aten) that needs to be linked in proper places (tests etc). Now that I stuck my finger in this, it involves extra cmake genuflexions, oh well..